File: src/ros_tasks/src/scripts/observer.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float32MultiArray
from std_msgs.msg import Float32

from sensor_msgs.msg import PointCloud
from geometry_msgs.msg import Point32


class Observer:

	def __init__(self):
		#subcsribe to the position topic that runner publishes to
		self.observer_sub = rospy.Subscriber('position', Float32MultiArray, self.callback)
		#use point cloud msg type to visually represent the path of the runner
		self.path_msg = PointCloud()
		self.path_pub = rospy.Publisher('path_topic', PointCloud, queue_size=10)
		self.path_msg.header.frame_id = "base_link"

	def callback(self, coords):
		#Create Point32 object from coordinates and add to PointCloud message
		#PointCloud message stores all coordinates so far and displays them 
		x = coords.data[0]
		y = coords.data[1]
		#publish visual representation of runner's path
		self.path_msg.points.append(Point32(x,y,0))
		self.path_pub.publish(self.path_msg)

if __name__ == '__main__':
	rospy.init_node('observer')
	observer = Observer()
	rospy.spin()

File: src/ros_tasks/src/scripts/observer.cpp
#include <iostream>
#include "ros/ros.h"
#include "std_msgs/Float32MultiArray.h"
#include <string>

/**
 * This tutorial demonstrates simple receipt of messages over the ROS system.
 */
using namespace std;
void observerCallback(const std_msgs::Float32MultiArray::ConstPtr& msg)
{
//  msg->data;
	std::cout<< "[" << msg->data[0] << ", " << msg->data[1] <<  ", "  << msg->data[2] << "]" << std::endl;

}

int main(int argc, char **argv)
{
  
  ros::init(argc, argv, "observer");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("position", 1000, observerCallback);
  ros::spin();

  return 0;
}